include/mesh/distributed_mesh.h

```diff
@@ -81,7 +81,27 @@ class DistributedMesh : public UnstructuredMesh
    * Copy and move assignment are not allowed.
    */
   DistributedMesh & operator= (const DistributedMesh &) = delete;
-  DistributedMesh & operator= (DistributedMesh &&) = delete;
+
+  /**
+   * Overloaded operator= will move contents of other_mesh to calling
+   * DistributedMesh object.
+   */
+  DistributedMesh & operator= (DistributedMesh && other_mesh);
+
+  /**
+   * Shim to call the move assignment operator for this class
```

Comment:
This comment also needs an update and the shim also ought to be moved to the .C file